src/core/radio/trel_interface.cpp

```diff
@@ -34,101 +34,415 @@
+    otLogDebgMac("Trel: Extended Address changed, re-registering DNS-SD service");
+    mRegisterServiceTask.Post();
+
+exit:
+    return;
+}
+
+void Interface::HandleExtPanIdChange(void)
+{
+    VerifyOrExit(mInitialized && mEnabled);
+    otLogDebgMac("Trel: Extended PAN ID changed, re-registering DNS-SD service");
+    mRegisterServiceTask.Post();
+
+exit:
+    return;
+}
+
+void Interface::HandleRegisterServiceTask(Tasklet &aTasklet)
+{
+    aTasklet.Get<Interface>().RegisterService();
+}
+
+void Interface::RegisterService(void)
+{
+    // TXT data consists of two entries: the "key" string, "=" char
+    // and length field (2 bytes) and 16 bytes for hex string
+    // representation of the 8 bytes MAC and Extended PAN ID values.
+    static constexpr uint8_t kTxtDataSize =
+        sizeof(kTxtRecordExtAddressKey) - 1 + sizeof(kTxtRecordExtPanIdKey) - 1 + (2 + 16) * 2;
+
+    uint8_t              txtData[kTxtDataSize];
+    uint8_t *            txtPtr = &txtData[0];
+    String<kTxtDataSize> string;
+
+    string.Append("%s=", kTxtRecordExtAddressKey);
+    string.AppendHexBytes(Get<Mac::Mac>().GetExtAddress().m8, sizeof(Mac::ExtAddress));
+
+    *txtPtr++ = static_cast<uint8_t>(string.GetLength());
+    memcpy(txtPtr, string.AsCString(), string.GetLength());
+    txtPtr += string.GetLength();
+
+    string.Clear();
+    string.Append("%s=", kTxtRecordExtPanIdKey);
+    string.AppendHexBytes(Get<Mac::Mac>().GetExtendedPanId().m8, sizeof(Mac::ExtendedPanId));
+
+    *txtPtr++ = static_cast<uint8_t>(string.GetLength());
+    memcpy(txtPtr, string.AsCString(), string.GetLength());
+    txtPtr += string.GetLength();
+
+    OT_ASSERT(txtPtr == OT_ARRAY_END(txtData));
+
+    otLogInfoMac("Trel: Registering DNS-SD service: port:%u, txt:\"%s=%s, %s=%s\"", mUdpPort, kTxtRecordExtAddressKey,
+                 Get<Mac::Mac>().GetExtAddress().ToString().AsCString(), kTxtRecordExtPanIdKey,
+                 Get<Mac::Mac>().GetExtendedPanId().ToString().AsCString());
+
+    otPlatTrelRegisterService(&GetInstance(), mUdpPort, txtData, sizeof(txtData));
+}
+
+extern "C" void otPlatTrelHandleDiscoveredPeerInfo(otInstance *aInstance, const otPlatTrelPeerInfo *aInfo)
+{
+    Instance &instance = AsCoreType(aInstance);
+
+    VerifyOrExit(instance.IsInitialized());
+    instance.Get<Interface>().HandleDiscoveredPeerInfo(*static_cast<const Interface::Peer::Info *>(aInfo));
+
+exit:
+    return;
+}
+
+void Interface::HandleDiscoveredPeerInfo(const Peer::Info &aInfo)
+{
+    Peer *             entry;
+    Mac::ExtAddress    extAddress;
+    Mac::ExtendedPanId extPanId;
+    bool               isNew = false;
+
+    VerifyOrExit(mInitialized && mEnabled);
+
+    SuccessOrExit(PrasePeerInfoTxtData(aInfo, extAddress, extPanId));
+
+    VerifyOrExit(extAddress != Get<Mac::Mac>().GetExtAddress());
+
+    if (aInfo.IsRemoved())
+    {
+        entry = mPeerTable.FindMatching(extAddress);
+        VerifyOrExit(entry != nullptr);
+        RemovePeerEntry(*entry);
```

Comment:
I think it may be better to leave it up to DNS-SD platform implementation to manage the discovered service entries and their TTL (I think each implementation may have its own mechanism for this). If an entry is timed out, the DNS-SD platform implementation can signal to `Trel::Interface` to remove the peer.

Reply:
Make sense to me. mDNS should signal an service instance removal if TTL runs out.